cmake-modules/AddTestCMocka.cmake

```diff
@@ -77,7 +77,21 @@ function(ADD_CMOCKA_TEST _TARGET_NAME)
     endif()
 
     if(DEFINED ENV{VCPKG_ROOT} OR DEFINED ENV{VCPKG_INSTALLATION_ROOT})
-        set(CMOCKA_LIB ${CMOCKA_LIBRARIES})
+        if(DEFINED ENV{VCPKG_ROOT})
+            set(VCPKG_PATH $ENV{VCPKG_ROOT})
+        endif()
+        if(DEFINED ENV{VCPKG_INSTALLATION_ROOT})
+            set(VCPKG_PATH "$ENV{VCPKG_INSTALLATION_ROOT}/")
+        endif()
+
+        # Patch for new VCPKG cmocka returning relative path
```

Comment:
Will we revert this once VCPKG issue is fixed (or maybe after the version pinning changes that @danieljurek was considering)?

Reply:
yeah, this is a temporal fix. 
It is setting the path to cmocka libraries manually and hardcoded.  This fix would work always while the name of the static library for cmocka doesn't change.

I am not sure if there is a bug for VCPKG, it might be that it used to be one and they fixed it and we were just making it work by luck wrongly before.

but, at any way, we need follow up on this

Reply:
Note from my message thread on Teams. I notified the person which made the change that there was an issue [here](https://github.com/microsoft/vcpkg/commit/1d8f0acc9c3085d18152a3f639077a28109196b6#r40299706). The issue is open [here](https://github.com/microsoft/vcpkg/issues/12193) and I'm working with here on the fix [here](https://github.com/microsoft/vcpkg/pull/12196)

Reply:
thank you @danewalton .
That's good news. 
I though cmocka could've done the change to work the same as when not using VCPKG. 
when not working with VCPKG, cmake gets the path to the libs in a relative way, just like it is now now.  That's why I thought that's how they wanted now.

So, as soon as VCPKG fix is merged, we can revert the changes here and it should work the same (and better without the hardcoded hot fix)